src/v2/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx

```diff
@@ -174,19 +187,39 @@ export const ArtworkFilterContextProvider: React.FC<
     initialFilterState
   )
 
+  const [stagedArtworkFilterState, stage] = useReducer(artworkFilterReducer, {})
+
   // TODO: Consolidate this into additional reducer
   const [filterAggregations, setAggregations] = useState(aggregations)
   const [artworkCounts, setCounts] = useState(counts)
+  const [shouldStageFilterChanges, setShouldStageFilterChanges] = useState(
+    false
+  )
 
   useDeepCompareEffect(() => {
     if (onChange) {
       onChange(omit(artworkFilterState, ["reset"]))
     }
   }, [artworkFilterState])
 
+  // If in staged mode, return the staged filters for UI display
+  const currentlySelectedFilters = () => {
+    return shouldStageFilterChanges
+      ? stagedArtworkFilterState
+      : artworkFilterState
+  }
+
+  // If in staged mode, manipulate the staged version of filter state
+  // instead of "real" one
+  const dispatchOrStage = (action: ArtworkFiltersAction) => {
+    shouldStageFilterChanges ? stage(action) : dispatch(action)
+  }
```

Comment:
Filter interactions get dispatched to either the real filters or the staged ones, depending on what mode we are in.